lisa/tools/hyperv.py

```diff
@@ -293,6 +293,37 @@ def create_nat(self, name: str, ip_range: str) -> None:
             force_run=True,
         )
 
+    def add_nat_mapping(
+        self, nat_name: str, internal_ip: str, external_port: int
+    ) -> None:
+        # delete existing NAT mapping
+        self.delete_nat_mapping(external_port)
```

Comment:
maintain the assigned external ports in the hyperV tool.
1. It can still set the start port number, but the default value is 50000.
2. In the hyperV tool, it saves assigned ports.
3. When it needs to assign a new port, start from the loweest unassigned port number.